atomwrappers.py

```diff
@@ -16,6 +17,8 @@ def __init__(self, name):
         self.name = name
 
     def __str__(self):
+        if self.atom_type == "GeneNode":
```

Comment:
A better place for the check would be GeneNode constructor. Or even better to make a factory function "make_gene_node" and place all the checks there. __str__  shouldn't modify the state of the object.

Reply:
I moved the check to the constructor